### Import

In [36]:
%matplotlib inline
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")
warnings.simplefilter(action='ignore', category=FutureWarning)

### Exploring Data

Data of this project include:  
- Unicorn company around the world as of August, 2020 (*__CB-Insights_Global-Unicorns_2020.csv__*).  
- Indian Startups and Investors (*__starup_funding.csv__*)
- Startup fundraising dataset (*__funds.csv__*)



We will analize distribution of unicorns,the top investors in each fields and the most popular investment trends.  
Analyze Indian startups and draw conclusions.

## Read csv file

In [2]:
unicorn_df = pd.read_csv("CB-Insights_Global-Unicorns_2020.csv")
startup_df = pd.read_csv("startup_funding.csv")

## Data overview

### Unicorns data

In [3]:
unicorn_df.head()

Unnamed: 0                Company Valuation ($B) Date Joined  \
0           0    Toutiao (Bytedance)           $140    4/7/2017   
1           1         Didi Chuxing              $56  12/31/2014   
2           2                 SpaceX            $46   12/1/2012   
3           3                 Stripe            $36   1/23/2014   
4           4  Palantir Technologies            $20    5/5/2011   

         Country                     Industry  \
0          China      Artificial intelligence   
1          China        Auto & transportation   
2  United States                        Other   
3  United States                      Fintech   
4  United States  Data management & analytics   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Matrix Partners, Tiger  Global Management, Sof...  
2  Founders Fund, Draper  Fisher Jurvetson, Rothe...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4             RRE Ventures, Founders  Fund, In-Q-Tel

In [4]:
unicorn_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 472 entries, 0 to 471
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Unnamed: 0        472 non-null    int64 
 1   Company           472 non-null    object
 2   Valuation ($B)    472 non-null    object
 3   Date Joined       472 non-null    object
 4   Country           472 non-null    object
 5   Industry          472 non-null    object
 6   Select Investors  471 non-null    object
dtypes: int64(1), object(6)
memory usage: 25.9+ KB


In [5]:
unicorn_df.nunique()

Unnamed: 0          472
Company             472
Valuation ($B)      117
Date Joined         401
Country              30
Industry             16
Select Investors    469
dtype: int64

In [6]:
unicorn_df.shape

(472, 7)

In [8]:
print("NaN")
print(unicorn_df.isnull().sum())

NaN
Unnamed: 0          0
Company             0
Valuation ($B)      0
Date Joined         0
Country             0
Industry            0
Select Investors    1
dtype: int64


### Getting to know the variables:  
- __Company__: Company's name
- __Valuation__: The value of a startup company (billion)  
- __Date joined__: Unicorn date  
- __Country__: The country where the unicorn is based
- __Category__: Company field
- __Select__ Investors: Investors for unicorns.


- Data has 449 rows and 6 columns:  
- Only 1 rows contains NaN value in "Select Investors" columns.
- 449 unicorns is recorgnized as of 1/29/2020.  
- 27 countries has unicorns.  
- 16 fields is selected and became unicorns.  
- The data type of 6 columns is object. But the data type of "valuation" columns should be Numerical, "Date joined" column should be datetime.

### Data Preprocessing

In [11]:
unicorn_df= unicorn_df.rename(columns = {'Unnamed': 'Rank'})
unicorn_df.head()

Rank                Company Valuation ($B) Date Joined        Country  \
0     0    Toutiao (Bytedance)           $140    4/7/2017          China   
1     1         Didi Chuxing              $56  12/31/2014          China   
2     2                 SpaceX            $46   12/1/2012  United States   
3     3                 Stripe            $36   1/23/2014  United States   
4     4  Palantir Technologies            $20    5/5/2011  United States   

                      Industry  \
0      Artificial intelligence   
1        Auto & transportation   
2                        Other   
3                      Fintech   
4  Data management & analytics   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Matrix Partners, Tiger  Global Management, Sof...  
2  Founders Fund, Draper  Fisher Jurvetson, Rothe...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4             RRE Ventures, Founders  Fund, In-Q-Tel

###  Valuadation columns  
- Deleted '$' and convert to numerical

In [12]:
unicorn_df = unicorn_df.replace('\$', '', regex=True)
unicorn_df["Valuation ($B)"] = unicorn_df["Valuation ($B)"].astype(float)

In [16]:
unicorn_df["Valuation ($B)"].describe()

count    472.000000
mean       3.155805
std        7.800915
min        1.000000
25%        1.000000
50%        1.500000
75%        2.800000
max      140.000000
Name: Valuation ($B), dtype: float64

### Date joined column  
- Converted to datetime

In [37]:
unicorn_df['Date Joined'] = pd.to_datetime(unicorn_df['Date Joined'])
unicorn_df['Date Joined'].describe()

count                     472
unique                    397
top       2018-11-13 00:00:00
freq                        4
first     2010-12-14 00:00:00
last      2020-08-14 00:00:00
Name: Date Joined, dtype: object

### Data after preprocessing

In [38]:
unicorn_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 472 entries, 0 to 471
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Rank              472 non-null    int64         
 1   Company           472 non-null    object        
 2   Valuation ($B)    472 non-null    float64       
 3   Date Joined       472 non-null    datetime64[ns]
 4   Country           472 non-null    object        
 5   Industry          472 non-null    object        
 6   Select Investors  471 non-null    object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(4)
memory usage: 25.9+ KB


In [19]:
unicorn_df.head()

Rank                Company  Valuation ($B) Date Joined        Country  \
0     0    Toutiao (Bytedance)           140.0  2017-04-07          China   
1     1         Didi Chuxing              56.0  2014-12-31          China   
2     2                 SpaceX            46.0  2012-12-01  United States   
3     3                 Stripe            36.0  2014-01-23  United States   
4     4  Palantir Technologies            20.0  2011-05-05  United States   

                      Industry  \
0      Artificial intelligence   
1        Auto & transportation   
2                        Other   
3                      Fintech   
4  Data management & analytics   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Matrix Partners, Tiger  Global Management, Sof...  
2  Founders Fund, Draper  Fisher Jurvetson, Rothe...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4             RRE Ventures, Founders  Fund, In-Q-Tel

## Indian Startups Data

### Meaning of data

In [20]:
startup_df.head()

Sr No Date dd/mm/yyyy                  Startup Name    Industry Vertical  \
0      1      09/01/2020                        BYJU’S               E-Tech   
1      2      13/01/2020                        Shuttl       Transportation   
2      3      09/01/2020                     Mamaearth           E-commerce   
3      4      02/01/2020  https://www.wealthbucket.in/              FinTech   
4      5      02/01/2020                        Fashor  Fashion and Apparel   

                             SubVertical City  Location  \
0                             E-learning      Bengaluru   
1              App based shuttle service        Gurgaon   
2  Retailer of baby and toddler products      Bengaluru   
3                      Online Investment      New Delhi   
4            Embroiled Clothes For Women         Mumbai   

              Investors Name       InvestmentnType Amount in USD Remarks  
0    Tiger Global Management  Private Equity Round  20,00,00,000     NaN  
1  Susquehanna Growth Equity              Series C     80,48,394     NaN  
2      Sequoia Capital India              Series B   1,83,58,860     NaN  
3             Vinod Khatumal          Pre-series A     30,00,000     NaN  
4    Sprout Venture Partners            Seed Round     18,00,000     NaN

In [21]:
startup_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3044 entries, 0 to 3043
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Sr No              3044 non-null   int64 
 1   Date dd/mm/yyyy    3044 non-null   object
 2   Startup Name       3044 non-null   object
 3   Industry Vertical  2873 non-null   object
 4   SubVertical        2108 non-null   object
 5   City  Location     2864 non-null   object
 6   Investors Name     3020 non-null   object
 7   InvestmentnType    3040 non-null   object
 8   Amount in USD      2084 non-null   object
 9   Remarks            419 non-null    object
dtypes: int64(1), object(9)
memory usage: 237.9+ KB


In [22]:
startup_df.nunique()

Sr No                3044
Date dd/mm/yyyy      1035
Startup Name         2459
Industry Vertical     821
SubVertical          1942
City  Location        112
Investors Name       2412
InvestmentnType        55
Amount in USD         471
Remarks                72
dtype: int64

In [23]:
print("NaN")
print(startup_df.isnull().sum())

NaN
Sr No                   0
Date dd/mm/yyyy         0
Startup Name            0
Industry Vertical     171
SubVertical           936
City  Location        180
Investors Name         24
InvestmentnType         4
Amount in USD         960
Remarks              2625
dtype: int64


In [24]:
startup_df.shape

(3044, 10)

In [32]:
startup_df = startup_df.drop('Remarks', axis=1)

### Getting to know the variables:  
- __Date__: Date of investment  
- __Startup__ Name: Company name  
- __Industry Vertical__: Main Business field of company  
- __SubVertical__: Sub business field of company  
- __City Location__: The city where the company's headquarters is located  
- __Investors Name__: Investors name  
- __Investors Type__: Type of investment  
- __Amount in USD__: Total amount invested

## Data Preprocessing

__Date column__  
- Covert to datetime  
Find and replace invalid date in Date columns (dd/mm/yyyy) to valid date.

In [40]:
data = startup_df['Date dd/mm/yyyy'].values

# Create dataframe
df = pd.DataFrame({'Date': data})

# Add 'corect' column to evalue 'Date' column corect or not
df['correct'] = pd.to_datetime(df['Date'],errors='coerce') #If 'coerce', then invalid parsing will be set as NaT.
df

Date    correct
0     09/01/2020 2020-09-01
1     13/01/2020 2020-01-13
2     09/01/2020 2020-09-01
3     02/01/2020 2020-02-01
4     02/01/2020 2020-02-01
...          ...        ...
3039  29/01/2015 2015-01-29
3040  29/01/2015 2015-01-29
3041  30/01/2015 2015-01-30
3042  30/01/2015 2015-01-30
3043  31/01/2015 2015-01-31

[3044 rows x 2 columns]

In [42]:
#Invalid value list
errors = df.loc[df['correct'].isnull()]['Date'].tolist()
errors

['05/072018', '01/07/015', '\\\\xc2\\\\xa010/7/2015', '22/01//2015']

In [43]:
startup_df['Date dd/mm/yyyy'].replace((['05/072018', '01/07/015', '\\\\xc2\\\\xa010/7/2015', '22/01//2015']),
                                        ('05/07/2018', '01/07/2015', '22/01/2015', '10/07/2015'), inplace=True)

In [45]:
startup_df['Date dd/mm/yyyy'] = pd.to_datetime(startup_df['Date dd/mm/yyyy'])

In [46]:
startup_df['Date dd/mm/yyyy'].describe()

count                    3044
unique                   1030
top       2015-08-07 00:00:00
freq                       11
first     2015-01-05 00:00:00
last      2020-10-01 00:00:00
Name: Date dd/mm/yyyy, dtype: object

As we can see:  
- We have 3044 rows, only 1030 unique values.  
- Successful fundraising rounds from 01/05/2015 to 10/01/2020

__Amount in USD column__

- Convert to float, handle invalid data   

In [48]:
startup_df['Amount in USD'] = startup_df["Amount in USD"].replace(",", "")
startup_df['Amount in USD'] = startup_df["Amount in USD"].replace("+", "")

In [51]:
startup_df['Amount in USD'] = startup_df['Amount in USD'].apply(lambda x: str(str(x).replace(",", "").replace("+", "")))

In [52]:
startup_df = startup_df.replace('\\\\\\\\xc2\\\\\\\\xa0','', regex=True)

In [53]:
startup_df['Amount in USD'].replace((['undisclosed', 'unknown','Undisclosed', 'N/A']),
                         ('NaN','NaN','NaN','NaN'), inplace=True)

In [54]:
startup_df['Amount in USD'] = startup_df['Amount in USD'].apply(lambda x: float(x))

In [55]:
startup_df['Amount in USD'].describe()

count    2.073000e+03
mean     1.840034e+07
std      1.211407e+08
min      1.600000e+04
25%      4.860000e+05
50%      1.750000e+06
75%      8.000000e+06
max      3.900000e+09
Name: Amount in USD, dtype: float64

### Data after preprocessing

In [56]:
startup_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3044 entries, 0 to 3043
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Sr No              3044 non-null   int64         
 1   Date dd/mm/yyyy    3044 non-null   datetime64[ns]
 2   Startup Name       3044 non-null   object        
 3   Industry Vertical  2873 non-null   object        
 4   SubVertical        2108 non-null   object        
 5   City  Location     2864 non-null   object        
 6   Investors Name     3020 non-null   object        
 7   InvestmentnType    3040 non-null   object        
 8   Amount in USD      2073 non-null   float64       
dtypes: datetime64[ns](1), float64(1), int64(1), object(6)
memory usage: 214.2+ KB


In [57]:
startup_df.head()

Sr No Date dd/mm/yyyy                  Startup Name    Industry Vertical  \
0      1      2020-09-01                        BYJU’S               E-Tech   
1      2      2020-01-13                        Shuttl       Transportation   
2      3      2020-09-01                     Mamaearth           E-commerce   
3      4      2020-02-01  https://www.wealthbucket.in/              FinTech   
4      5      2020-02-01                        Fashor  Fashion and Apparel   

                             SubVertical City  Location  \
0                             E-learning      Bengaluru   
1              App based shuttle service        Gurgaon   
2  Retailer of baby and toddler products      Bengaluru   
3                      Online Investment      New Delhi   
4            Embroiled Clothes For Women         Mumbai   

              Investors Name       InvestmentnType  Amount in USD  
0    Tiger Global Management  Private Equity Round    200000000.0  
1  Susquehanna Growth Equity              Series C      8048394.0  
2      Sequoia Capital India              Series B     18358860.0  
3             Vinod Khatumal          Pre-series A      3000000.0  
4    Sprout Venture Partners            Seed Round      1800000.0

## Exploring Data

### Review Data

In [58]:
unicorn_df.head()

Rank                Company  Valuation ($B) Date Joined        Country  \
0     0    Toutiao (Bytedance)           140.0  2017-04-07          China   
1     1         Didi Chuxing              56.0  2014-12-31          China   
2     2                 SpaceX            46.0  2012-12-01  United States   
3     3                 Stripe            36.0  2014-01-23  United States   
4     4  Palantir Technologies            20.0  2011-05-05  United States   

                      Industry  \
0      Artificial intelligence   
1        Auto & transportation   
2                        Other   
3                      Fintech   
4  Data management & analytics   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Matrix Partners, Tiger  Global Management, Sof...  
2  Founders Fund, Draper  Fisher Jurvetson, Rothe...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4             RRE Ventures, Founders  Fund, In-Q-Tel

In [59]:
startup_df.head()

Sr No Date dd/mm/yyyy                  Startup Name    Industry Vertical  \
0      1      2020-09-01                        BYJU’S               E-Tech   
1      2      2020-01-13                        Shuttl       Transportation   
2      3      2020-09-01                     Mamaearth           E-commerce   
3      4      2020-02-01  https://www.wealthbucket.in/              FinTech   
4      5      2020-02-01                        Fashor  Fashion and Apparel   

                             SubVertical City  Location  \
0                             E-learning      Bengaluru   
1              App based shuttle service        Gurgaon   
2  Retailer of baby and toddler products      Bengaluru   
3                      Online Investment      New Delhi   
4            Embroiled Clothes For Women         Mumbai   

              Investors Name       InvestmentnType  Amount in USD  
0    Tiger Global Management  Private Equity Round    200000000.0  
1  Susquehanna Growth Equity              Series C      8048394.0  
2      Sequoia Capital India              Series B     18358860.0  
3             Vinod Khatumal          Pre-series A      3000000.0  
4    Sprout Venture Partners            Seed Round      1800000.0

## Answer the questions
__Unicorns data__
- Number of unicorns by country, which country has the most unicorns.  
- Which field has the most unicorns. Sort by company's value. Top 5 fields with the highest value.  
- Which investors have ownership the most unicorns.  
- The numbers of unicorns by years. Which years has the most unicorns.  
- Statistics in Southeast Asia. 

__Startup data in India__  
- Which field has the most unicorns in India. Is it similar to the trend of the world?  
- The distribution of SubVertical.  
- Subvertical distribution of top 3 Industry Vertical.  
- Distribution of funding in India. 
- Startups, fields with potential for investment. 
- Amount invested over time.
- Which startups have the most investment rounds in India.

## Data Processing

__Unicorns data__

Add new clolumns such as Investor_1, Investor_2,.. from Investor columns.  
Handling data inconsistency such as ('.', ','), ('and', ','), Upper all of company name,..

In [60]:
unicorn_df.iloc[406]

Rank                                                          406
Company                                                     Loggi
Valuation ($B)                                                1.0
Date Joined                                   2019-06-05 00:00:00
Country                                                    Brazil
Industry                      Supply chain, logistics, & delivery
Select Investors    Qualcomm Ventures, SoftBank Group. Monashees+
Name: 406, dtype: object

In [61]:
unicorn_df.loc[unicorn_df['Rank'] == 406, 'Select Investors'] = 'Qualcomm Ventures, SoftBank Group, Monashees+'

In [62]:
unicorn_df.iloc[1]

Rank                                                                1
Company                                                Didi Chuxing  
Valuation ($B)                                                   56.0
Date Joined                                       2014-12-31 00:00:00
Country                                                         China
Industry                                        Auto & transportation
Select Investors    Matrix Partners, Tiger  Global Management, Sof...
Name: 1, dtype: object

In [63]:
unicorn_df.loc[unicorn_df['Rank'] == 1, 'Select Investors'] = 'Matrix Partners, Tiger  Global Management, Sofbank Corp.'

In [64]:
unicorn_df = unicorn_df.replace('and','', regex=True)

In [66]:
unicorn_df[unicorn_df['Select Investors'].str.contains("SoftBankGroup", na=False)]

Rank           Company  Valuation ($B) Date Joined        Country  \
27    27  Roivant Sciences            9.09  2018-11-13  United States   
34    34          Snapdeal            7.00  2014-05-21          India   
35    35         Tokopedia            7.00  2018-12-12      Indonesia   

                           Industry  \
27                           Health   
34  E-commerce & direct-to-consumer   
35  E-commerce & direct-to-consumer   

                                     Select Investors  
27                       SoftBankGroup, Founders Fund  
34            SoftBankGroup, Blackrock, Alibaba Group  
35  SoftBankGroup, Alibaba Group, Sequoia Capital ...

In [69]:
unicorn_df = unicorn_df.replace('SoftBankGroup', 'Sortbank Group', regex=True)

In [72]:
df = pd.concat([unicorn_df['Select Investors'].str.split(',', expand=True)], axis=1)
df

0                            1  \
0    Sequoia Capital China         SIG Asia Investments   
1          Matrix Partners      Tiger Global Management   
2            Founders Fund     Draper  Fisher Jurvetson   
3          Khosla Ventures             LowercaseCapital   
4             RRE Ventures               Founders  Fund   
..                     ...                          ...   
467                   Coty                         None   
468                Upper90           RiverPark Ventures   
469           WRVI Capital            Qualcomm Ventures   
470    One Equity Partners                         None   
471          T. Rowe Price   Dragoneer Investment Group   

                         2                3  
0               Sina Weibo   Softbank Group  
1            Sofbank Corp.             None  
2      Rothenberg Ventures             None  
3                 capitalG             None  
4                 In-Q-Tel             None  
..                     ...              ...  
467                   None             None  
468   Advent International             None  
469        S-Cubed Capital             None  
470                   None             None  
471              BlackRock             None  

[472 rows x 4 columns]

In [78]:
df = pd.concat([unicorn_df['Select Investors'].str.split(',', expand=True)], axis=1)
unicorn_df = unicorn_df.merge(df)
unicorn_df = unicorn_df.rename(columns = {0:'Investor_1', 1: 'Investor_2', 2: 'Investor_3', 3: 'Investor_4'})
unicorn_df['Investor_1'] = unicorn_df['Investor_1'].str.strip()
unicorn_df['Investor_2'] = unicorn_df['Investor_2'].str.strip()
unicorn_df['Investor_3'] = unicorn_df['Investor_3'].str.strip()
unicorn_df['Investor_4'] = unicorn_df['Investor_4'].str.strip()

unicorn_df['Investor_1'] = unicorn_df['Investor_1'].str.upper()
unicorn_df['Investor_2'] = unicorn_df['Investor_2'].str.upper()
unicorn_df['Investor_3'] = unicorn_df['Investor_3'].str.upper()
unicorn_df['Investor_4'] = unicorn_df['Investor_4'].str.upper()
unicorn_df.head()

Rank                Company  Valuation ($B) Date Joined        Country  \
0     0    Toutiao (Bytedance)           140.0  2017-04-07          China   
1     1         Didi Chuxing              56.0  2014-12-31          China   
2     2                 SpaceX            46.0  2012-12-01  United States   
3     3                 Stripe            36.0  2014-01-23  United States   
4     4  Palantir Technologies            20.0  2011-05-05  United States   

                      Industry  \
0      Artificial intelligence   
1        Auto & transportation   
2                        Other   
3                      Fintech   
4  Data management & analytics   

                                    Select Investors             Investor_1  \
0  Sequoia Capital China, SIG Asia Investments, S...  SEQUOIA CAPITAL CHINA   
1  Matrix Partners, Tiger Global Management, Sofb...        MATRIX PARTNERS   
2  Founders Fund, Draper  Fisher Jurvetson, Rothe...          FOUNDERS FUND   
3        Khosla Ventures, LowercaseCapital, capitalG        KHOSLA VENTURES   
4             RRE Ventures, Founders  Fund, In-Q-Tel           RRE VENTURES   

                 Investor_2           Investor_3      Investor_4  
0      SIG ASIA INVESTMENTS           SINA WEIBO  SOFTBANK GROUP  
1   TIGER GLOBAL MANAGEMENT        SOFBANK CORP.            None  
2  DRAPER  FISHER JURVETSON  ROTHENBERG VENTURES            None  
3          LOWERCASECAPITAL             CAPITALG            None  
4            FOUNDERS  FUND             IN-Q-TEL            None

## Answer the questions
__Unicorns data__
- Number of unicorns by country, which country has the most unicorns.  
- Which field has the most unicorns. Sort by company's value. Top 5 fields with the highest value.  
- Which investors have ownership the most unicorns.  
- The numbers of unicorns by years. Which years has the most unicorns.  
- Statistics in Southeast Asia. 

__Number of unicorns by country, which country has the most unicorns?__

In [ ]:
country_unicorns = unicorn_df['Country'].value_counts()
